assignments/this.js

```diff
@@ -1,26 +1,81 @@
-/* The for principles of "this";
+/* The four principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+1.New Binding: Is the function called by new?
+3.Implicit Binding: Object Literals - Is the function called as a method, ie: obj.func()
+2.Explicit Binding: Is the function called by call(), apply(), or bind()?
+4.Window
+
+If strict mode is enabled, return undefined.
+Otherwise, return the global object, ie: window.
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
 
-// code example for Window Binding
+// code example for New Binding
+
+function Animal(food) {
+    this.food = food;
+    this.eat = function() {
+      console.log(`This animal likes to eat ${this.food}`);
+    }
+  }
+
+  const zebra = new Animal('grass'); //new keyword used to bind this keyword
 
 // Principle 2
 
 // code example for Implicit Binding
+const hobbit = {
+      name: "Samwise",
+      age: 32,
+      food: "taters",
+      cook: function() {
+        return `${this.name} loves to cook ${this.food}`;
+      }
+    }
+    
+    // Look at where the method was invoked and you will find your binding!
+    console.log(hobbit.cook());
+    //hobbit acts are this keyword 
 
 // Principle 3
 
-// code example for New Binding
+// code example for Explicit Binding
+const person = {
+      name: "Jill"
+    }
+    
+    const hobbit = {
+      name: "Frodo"
+    }
+    
+    const skills = ["HTML","CSS","JS"];
+    
+    function introduce(skills1, skills2, skills3) {
+      debugger;
+      return `Hello! my name is: ${this.name} and these are my skills: ${skills1}, ${skills2}, ${skills3}`;
+    }
+    
+    // Fucntion called with .call() - Passes arguments individually
+    console.log(introduce.call(person, skills));
```

Comment:
right now this will print `Hello! my name is: Jill and these are my skills: HTML,CSS,JS, undefined, undefined` because when you passed in your skills array `call()` saw that as a single argument and passed it into `introduce()` as skills1, but nothing passed to skills2 or 3. if you use the spread operator in call it will open up that array and pass in the three items in it `console.log(introduce.call(person, ...skills))`, then it'll work as you expect

Reply:
ah ok. I having this issue with my flying rodent man revealing his weapons and I was wondering why I thought that syntax was correct I missed the  . . . and thats `dot space dot space dot`, right?